src/fmu/dataio/_objectdata_provider.py

```diff
@@ -622,6 +677,7 @@ def derive_metadata(self):
         meta["depth_reference"] = list(self.dataio.vertical_domain.values())[0]
         meta["spec"] = objres["spec"]
         meta["bbox"] = objres["bbox"]
+        meta["table_index"] = objres["table_index"]
```

Comment:
@daniel-sol I think the error in pytest are related to this line. From what I can see, `objres` dictionary will have `table_index` key only if the `objres` is a pandas.dataframe and/or pyarrow table

if you change it to this block below, I think the code should pass the test. But, I am a bit confused on how it pass the test when you are running them on different location.

```
        if "table_index" in objres:
            meta["table_index"] = objres["table_index"]
```

Reply:
I changed it to `objres.get("table_index")` instead, which will default to `None` if it doesn't exist. So that will make the test pass. Also tests work for me from root, but I have to install dataio. With the `-e` option that is not a bit issue (for me).

Reply:
Thanks a lot. Things are working now. But only when installing with -e, something I don't commonly do.